### DISCUSSION TITLE
trap exception with mcause = 0x8000001F

### DISCUSSION THREAD

**Discussion by patricevenant** [Mi-V-Soft-RISC-V/miv-rv32-documentation#21] [Mi-V RV32 Soft Processors] [answered]
Hi,
I got a trap exception with mcause = 0x8000001F. I didn't find which cause it is :( In the Handbook (DS00003721A-page 26), the latest cause is 0x1E. I'm using a RISC-V core RV32I (misa = 0x42000100). All external interrupts are hard wired to 0 level.

Any clue ?
Thanks,
Patrice

**Comment by CLappin** [comment, marked as answer]
Hi @patricevenant ,

If you take a look at the RISC-V Privileged Spec (Chaper 3.1.16) the mcause register, that may point you to the possible issue.

Thanks. 
Ciaran

**Reply by patricevenant** [reply]
Thanks to point me on the documentation. The value 0x8000001F wasn't correct, I didn't see the correct value and saw a different register instead. The correct value of mcause was 0x00000018. Which is correct, the RISC-V jump to unknown memory. Software bug in this case.
Thanks for you help,
Patrice

**Reply by CLappin** [reply]
Hi Patrice, 

No problem. We're also working with Andre (Microchip FAE) internally to see what is going on. Hopefully we can figure this out. 

Thanks, 
Ciaran

**Reply by nitindeshpande** [reply]
> Thanks to point me on the documentation. The value 0x8000001F wasn't correct, I didn't see the correct value and saw a different register instead. The correct value of mcause was 0x00000018. Which is correct, the RISC-V jump to unknown memory. Software bug in this case. Thanks for you help, Patrice

Curious. Are you using the interrupts in the vectored mode?
